Nimble/Wrappers/MatcherFunc.swift

```diff
@@ -1,5 +1,13 @@
 import Foundation
 
+/// A convenience API to build matchers that allow full control over
+/// to() and toNot() match cases.
+///
+/// The final bool argument in the closure is if the match is for negation.
+///
+/// You may use this when implementing your own custom matchers.
+///
+/// But if you want to receive matchers, use the Matcher protocol instead.
```

Comment:
Nitpick: This sentence is a little confusing. I interpreted this to mean: "If your matcher takes other matchers as an argument, use MatcherFunc instead."